tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+
+    # Adding nodes, node-01 - node-06
+    url_node = 'http://127.0.0.1:8000/node/'
+    api_nodename = 'http://schema.massopencloud.org/haas/v0/obm/'
+
+    for i in range(1, 10):
+        obminfo = {
+                "type": api_nodename + 'ipmi', "host": "10.10.0.0"+repr(i),
+                "user": "ipmi_u", "password": "pass1234"
+                }
+        sess.put(
+                url_node + 'node-0'+repr(i), data=json.dumps({"obm": obminfo})
+                )
+        sess.put(
+                url_node + 'node-0' + repr(i) + '/nic/eth0', data=json.dumps(
+                            {"macaddr": "aa:bb:cc:dd:ee:0" + repr(i)}
+                            )
+                     )
+
+    # Adding Projects proj-01 - proj-03
+    for i in ["proj-01", "proj-02", "proj-03"]:
+        sess.put('http://127.0.0.1:8000/project/' + i)
+
+    # Adding switches one for each driver
+    url_switch = 'http://127.0.0.1:8000/switch/'
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/'
+
+    dell_param = {
+            'type': api_name + 'powerconnect55xx', 'hostname': 'dell-01',
+            'username': 'root', 'password': 'root1234'
+            }
+    nexus_param = {
+            'type': api_name + 'nexus', 'hostname': 'nexus-01',
+            'username': 'root', 'password': 'root1234', 'dummy_vlan': '333'
+            }
+    mock_param = {
+            'type': api_name + 'mock', 'hostname': 'mockSwitch-01',
+            'username': 'root', 'password': 'root1234'
+            }
+    brocade_param = {
+            'type': api_name + 'brocade', 'hostname': 'brocade-01',
+            'username': 'root', 'password': 'root1234',
+            'interface_type': 'TenGigabitEthernet'
+            }
+
+    sess.put(url_switch + 'dell-01', data=json.dumps(dell_param))
+    sess.put(url_switch + 'nexus-01', data=json.dumps(nexus_param))
+    sess.put(url_switch + 'mock-01', data=json.dumps(mock_param))
+    sess.put(url_switch + 'brocade-01', data=json.dumps(brocade_param))
+
+    # Adding ports to the mock switch, Connect nics to ports:
+    for i in range(1, 8):
+        sess.put(url_switch + 'mock-01/port/gi1/0/' + repr(i))
+        sess.post(
+                url_switch + 'mock-01/port/gi1/0/' + repr(i) + '/connect_nic',
+                data=json.dumps(
+                    {'node': 'node-0' + repr(i), 'nic': 'eth0'}
+                    )
+                )
+
+# Adding port gi1/0/8 to switch mock-01 without connecting it to any node.
+    sess.put(url_switch + 'mock-01/port/gi1/0/8')
+
+    # Adding Projects proj-01 - proj-03
+    for i in ["proj-01", "proj-02", "proj-03"]:
+        sess.put('http://127.0.0.1:8000/project/' + i)
+
+    # Allocating nodes to projects
+    url_project = 'http://127.0.0.1:8000/project/'
+    # Adding nodes 1 to proj-01
+    sess.post(
+            url_project + 'proj-01' + '/connect_node',
+            data=json.dumps({'node': 'node-01'})
+            )
+    # Adding nodes 2, 4 to proj-02
```

Comment:
Since each project was allocated different number of nodes, a for loop was not feasible. But I get your point. Hence wrote a function to assign variable number of nodes to a given project. Next commit will have that.